### ISSUE TITLE
Builds failing on Windows due to pytest upstream change

### ISSUE THREAD

**Issue by jat255** [hyperspy/hyperspy#1787]
See discussion in #1782.

A recent change between versions 3.2.5 and 3.3.0 in `pytest` has caused an error to popup in the `test_edax.py` test suite (see below). It appears that one of the files is not being released properly, and so when the temporary directory is cleaned up, Windows throws an error. 

This doesn't happen in the other two test suites that use `pytest.fixture` (`test_blockfile.py` and `test_hdf5.py`).

```
λ pytest -q hyperspy\tests\io\test_edax.py
..................                                                       [100%]E [105%]

=================================== ERRORS ====================================
__________ ERROR at teardown of TestSpdMap_061_xrf.test_spc_reading ___________

    @pytest.fixture(scope="module")
    def tmpdir():
        import zipfile
        zipf = os.path.join(MY_PATH, "edax_files.zip")
        with zipfile.ZipFile(zipf, 'r') as zipped:
            with tempfile.TemporaryDirectory() as tmp:
                zipped.extractall(tmp)
                # spd_fname = os.path.join(tmp, TEST_FILES[3])
                # print(os.listdir(tmpdir))
                # print(spd_fname)
                yield tmp
                # Force files release (required in Windows)
>               gc.collect()

hyperspy\tests\io\test_edax.py:38:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\..\appdata\local\programs\python\python35\lib\tempfile.py:808: in __exit__
    self.cleanup()
..\..\appdata\local\programs\python\python35\lib\tempfile.py:812: in cleanup
    _shutil.rmtree(self.name)
..\..\appdata\local\programs\python\python35\lib\shutil.py:488: in rmtree
    return _rmtree_unsafe(path, onerror)
..\..\appdata\local\programs\python\python35\lib\shutil.py:383: in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

path = 'C:\\Users\\jat\\AppData\\Local\\Temp\\1\\tmpy0fbmh02'
onerror = <function rmtree.<locals>.onerror at 0x0000025B3BC1B400>

    def _rmtree_unsafe(path, onerror):
        try:
            if os.path.islink(path):
                # symlinks to directories are forbidden, see bug #1669
                raise OSError("Cannot call rmtree on a symbolic link")
        except OSError:
            onerror(os.path.islink, path, sys.exc_info())
            # can't continue even if onerror hook returns
            return
        names = []
        try:
            names = os.listdir(path)
        except OSError:
            onerror(os.listdir, path, sys.exc_info())
        for name in names:
            fullname = os.path.join(path, name)
            try:
                mode = os.lstat(fullname).st_mode
            except OSError:
                mode = 0
            if stat.S_ISDIR(mode):
                _rmtree_unsafe(fullname, onerror)
            else:
                try:
>                   os.unlink(fullname)
E                   PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\jat\\AppData\\Local\\Temp\\1\\tmpy0fbmh02\\single_spect.spc'

..\..\appdata\local\programs\python\python35\lib\shutil.py:381: PermissionError
```

**Comment by ericpre** [comment]
Maybe an alternative would be to use the [teardown_class](https://docs.pytest.org/en/latest/xunit_setup.html?highlight=setup_class#method-and-function-level-setup-teardown) method? At the moment, this is unpacking the file and creating a temporary folder before running each test, if I read correctly the code? Also, it could make the test more straightforward to read possibly.

**Comment by jat255** [comment]
That's actually how I had originally coded it, but then it got switched to fixtures during the switch from `nosetests` to `pytest` (#1413 )

**Comment by jat255** [comment]
Closing, because this was resolved by #1788
